src/aiodynamo/client.py

```diff
@@ -486,94 +555,67 @@ def table(self, name: TableName) -> Table:
-            coro_func,
-            "ExclusiveStartKey",
-            "LastEvaluatedKey",
-            "Items",
-            py2dy(start_key) if start_key else None,
-            limit,
-            "Limit",
-        ):
-            yield dy2py(raw, self.numeric_type)
+            payload["FilterExpression"] = filter_expression.encode(params)
+        if projection or filter_expression:
+            payload["ExpressionAttributeNames"] = params.get_expression_names()
+            payload["ExpressionAttributeValues"] = params.get_expression_values()
+
+        got = 0
+        async for result in self._fast_depaginate("Scan", payload):
+            for item in result["Items"]:
+                yield dy2py(item, self.numeric_type)
+                if limit:
+                    got += 1
+                    if got >= limit:
+                        return
 
     async def count(
         self,
         table: TableName,
-        key_condition: ConditionBase,
+        key_condition: KeyCondition,
         *,
         start_key: Dict[str, Any] = None,
-        filter_expression: ConditionBase = None,
-        index: str = None
+        filter_expression: Condition = None,
+        index: str = None,
     ) -> int:
-        expression_attribute_names = {}
-        expression_attribute_values = {}
+        params = Parameters()
 
-        builder = ConditionExpressionBuilder()
+        payload = {
+            "TableName": table,
+            "KeyConditionExpression": key_condition.encode(params),
+            "Select": Select.count.value,
+        }
 
+        if start_key:
+            payload["ExclusiveStartKey"] = start_key
         if filter_expression:
-            filter_expression, ean, eav = builder.build_expression(filter_expression)
-            expression_attribute_names.update(ean)
-            expression_attribute_values.update(eav)
-
-        if key_condition:
-            key_condition_expression, ean, eav = builder.build_expression(
-                key_condition, True
-            )
-            expression_attribute_names.update(ean)
-            expression_attribute_values.update(eav)
-        else:
-            key_condition_expression = None
-
-        coro_func = partial(
-            self.core.query,
-            **clean(
-                TableName=table,
-                IndexName=index,
-                FilterExpression=filter_expression,
-                KeyConditionExpression=key_condition_expression,
-                ExpressionAttributeNames=expression_attribute_names,
-                ExpressionAttributeValues=py2dy(expression_attribute_values),
-                Select=Select.count.value,
-            )
-        )
+            payload["FilterExpression"] = filter_expression.encode(params)
+        if index:
+            payload["IndexName"] = index
+        payload["ExpressionAttributeNames"] = params.get_expression_names()
+        payload["ExpressionAttributeValues"] = params.get_expression_values()
```

Comment:
is it valid to include these elements if key condition and filter expression are both empty?

Reply:
`count` requires a Key Condition and Key Conditions are not allowed to be empty.